### PR TITLE
Feature/FileHandle improvements

### PR DESCRIPTION
**Pull request by JeppeBlixen**
## What type of PR is this? (check all applicable)

- Refactor
- Feature

### Overview
A number of general improvements to the FileHandler, all non breaking:
- fix: Set correct return of set_schema function
- feat: Add get_options and set_options function to TableHandle
- feat: Update schema_location to be optional parameter on FileHandle
- feat: Update FileHandle to use same options dict for batch and streaming
- refactor: Update fileHandle to use get instead of table_property on configurator
- feat: Remove this FileHandleTest as a second more general one exists

### What is the current behavior?
*If this PR addresses an existing issue, describe the current behavior. Otherwise, you can skip this section.* 

### What is the new behavior?
*Describe the new behavior or the desired outcome of the changes in this PR.*

### Does this PR introduce a breaking change?
*Are there any breaking changes introduced by this PR? If so, what changes might users need to make in their application due to this PR?*

## Related Issues (Link to the open issue)
*Provide a link to the open issue(s) related to this PR, if applicable*

## Additional information
*Provide any other relevant information, such as links to webpages, screenshots, or any other context that may help in understanding the changes in this PR.*

### CODE REVIEW

File: src/spetlr/filehandle/file_handle.py [hunk 1]
```diff
@@ -76,20 +91,26 @@ def read(self) -> DataFrame:
...
     def read_stream(self) -> DataFrame:
-        reader = (
-            Spark.get()
-            .readStream.format("cloudFiles")
-            .option("cloudFiles.format", self._data_format)
-            .option("cloudFiles.schemaLocation", self._schema_location)
-        )
+        reader = Spark.get().readStream.format("cloudFiles")
 
         if self._options is not None:
```
Comment (MartinBoge) [inline src/spetlr/filehandle/file_handle.py:96]: Here we can just say `if self._options` also refactor this for the `read` method. Just to stay consistent in the module.

File: src/spetlr/filehandle/file_handle.py [hunk 2]
```diff
@@ -76,20 +91,26 @@ def read(self) -> DataFrame:
...
+        if self._schema:
+            reader = reader.schema(self._schema)
+
         return reader.load(self._location)
 
     def get_schema(self) -> T.StructType:
         return self._schema
 
-    def set_schema(self, schema: T.StructType) -> T.StructType:
+    def set_schema(self, schema: T.StructType) -> "FileHandle":
```
Comment (MartinBoge) [inline src/spetlr/filehandle/file_handle.py:107]: Remember to change the set_schema to return `self`

File: src/spetlr/tables/TableHandle.py [hunk 3]
```diff
@@ -49,5 +49,11 @@ def get_table_id(self):
     def get_schema(self) -> T.StructType:
         raise NotImplementedError()
 
-    def set_schema(self, schema: T.StructType) -> T.StructType:
+    def set_schema(self, schema: T.StructType) -> "TableHandle":
```
Comment (MartinBoge) [inline src/spetlr/tables/TableHandle.py:52]: Just curious. 

1. Do we have a great need for these getters and setters? Personally, i couldn't see where i would use them. 

3. What is the reasoning behind the design choice of not using the `property` function/decorator with such behaviour?

**Review by MartinBoge — APPROVED** [review]
lgmt

File: src/spetlr/tables/TableHandle.py (hunk 3 above)
Comment (JeppeBlixen) [reply src/spetlr/tables/TableHandle.py:52]: 1. Yes. The get/set schema is used in the schema validation extractor. I want get/set options in cases where the handle is created from table config, but some extra options is set.
2. Could you give an example of what you mean by the property decorator.

**Pull request merged**